src/AFQMC/Estimators/BackPropagatedEstimator.hpp

```diff
@@ -188,14 +188,14 @@ class BackPropagatedEstimator : public EstimatorBase
       int nx((walker_type == COLLINEAR) ? 2 : 1);
 
       // 1. check structures
-      if (Refs.size(0) != wset.size() || Refs.size(1) != nrefs || Refs.size(2) != nrow * ncol)
+      if (std::get<0>(Refs.sizes()) != wset.size() || std::get<1>(Refs.sizes()) != nrefs || std::get<2>(Refs.sizes()) != nrow * ncol)
```

Comment:
I just feel the code is less readable. What is the gain of switching to compile time?

Reply:
Runtime access to dimension index is an was anomaly in Multi that is hard to maintain. Compile-time access also detects errors when compiling; and in fact detected an bug already in AFQMC. The changes look ugly because they were done mechanically (but surely).

Reply:
Possible to have
```
template<unsigned DIM>
size() { return std::get<DIM>(sizes()); }
```
so we can write `Refs.size<0>()`?

Reply:
Yes, it is quite awkward currently in C++ to have static index access. Supposedly in future versions of C++ there will be a `get` member (`m.sizes().get<D>()`). I am debating what to do. The problem with `m.size<D>()` is that `m.size()` is already a function. `D=0` could be the default but it is generate inconsistencies with other functions where the nice default is `D=1` (like `m.rotated()`). The other option is to have `m.sizes_at<D>()` but independently of that it generates a lot of duplicated code, like `m.extensions_at`, `m.strides_at`, `m.offsets_at`.